File: RentACar/UI/rentacar_mobile/lib/models/vozilo_pregled.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:rentacar_admin/models/vozila.dart';
part 'vozilo_pregled.g.dart';

@JsonSerializable()
class VoziloPregled {
  int? voziloPregledId;
  int? voziloId;
  DateTime? datum;
  Vozilo? vozilo;

  VoziloPregled({this.voziloPregledId, this.voziloId, this.datum, this.vozilo});

  /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
  factory VoziloPregled.fromJson(Map<String, dynamic> json) =>
      _$VoziloPregledFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$VoziloPregledToJson(this);
}

File: RentACarDodatni/RentACarDodatni/Database/VoziloPregled.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RentACar.Services.Database
{
    public class VoziloPregled
    {
        public int VoziloPregledId { get; set; }
        public int VoziloId { get; set; }
        public Vozila Vozilo { get; set; }
        public DateTime Datum { get; set; }
    }
}
